include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -0,0 +1,968 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the seqan3::alignment_file_format_bam class.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <iterator>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna15.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/detail/reflection.hpp>
+#include <seqan3/core/detail/to_string.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/io/alignment_file/detail.hpp>
+#include <seqan3/io/alignment_file/format_sam.hpp>
+#include <seqan3/io/alignment_file/header.hpp>
+#include <seqan3/io/alignment_file/input_options.hpp>
+#include <seqan3/io/alignment_file/output_options.hpp>
+#include <seqan3/io/alignment_file/sam_tag_dictionary.hpp>
+#include <seqan3/io/detail/ignore_output_iterator.hpp>
+#include <seqan3/io/detail/misc.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/slice.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/range/view/take_until.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+
+#include <seqan3/alphabet/detail/convert.hpp> // bam write
+#include <seqan3/range/view/single_pass_input.hpp>
+
+namespace seqan3
+{
+
+/*!\brief       The BAM format.
+ * \implements  alignment_file_format_concept
+ * \ingroup     alignment_file
+ *
+ * \details
+ *
+ * The BAM format is the binary version of the SAM format:
+ *
```

Comment:
maybe a link to the format specification? Did we do it for other formats?

Reply:
I copy all the SAM format details in a line below so I think a link redundant?